src/modules/hip/kernel/rain.hpp

```diff
@@ -0,0 +1,300 @@
+}
+
+template <typename T>
+RppStatus hip_exec_rain_tensor(T *srcPtr1,
+                               RpptDescPtr srcDescPtr,
+                               T *dstPtr,
+                               RpptDescPtr dstDescPtr,
+                               Rpp32f rainPercentage,
+                               Rpp32u rainWidth,
+                               Rpp32u rainHeight,
+                               Rpp32s slant,
+                               Rpp32f *alpha,
+                               RpptROIPtr roiTensorPtrSrc,
+                               RpptRoiType roiType,
+                               rpp::Handle& handle)
+{
+    if (roiType == RpptRoiType::LTRB)
+        hip_exec_roi_converison_ltrb_to_xywh(roiTensorPtrSrc, handle);
+
+    Rpp32f rainPercent = rainPercentage * 0.004f;
+    Rpp32u numDrops = static_cast<Rpp32u>(rainPercent * srcDescPtr->h * srcDescPtr->w);
+    std::srand(std::time(0));
+    T *rainLayer = reinterpret_cast<T *>(handle.GetInitHandle()->mem.mgpu.scratchBufferPinned.floatmem);
+    T initValue;
+    if constexpr (std::is_same<T, Rpp8s>::value)
+    {
+        initValue = static_cast<T>(0x81);
+    }
+    else
+    {
+        initValue = static_cast<T>(0);
+    }
+    hipMemset(rainLayer, initValue, srcDescPtr->strides.nStride * sizeof(T));
+    for (Rpp32u i = 0; i < numDrops; i++)
+    {
+        Rpp32u xStart = rand() % (srcDescPtr->w - slant);
+        Rpp32u yStart = rand() % (srcDescPtr->h - rainHeight);
+        for (Rpp32u j = 0; j < rainHeight; j++) // height - rainHeight
+        {
+            Rpp32u x = xStart + j * slant / rainHeight;
+            Rpp32u y = yStart + j;
+
+            if (x >= 0 && x < srcDescPtr->w && y < srcDescPtr->h)
+            {
+                Rpp32s loc = y * srcDescPtr->strides.hStride + x * srcDescPtr->strides.wStride;
+                T *rainLayerTemp = rainLayer + loc;
+
+                // Conditionally assign rain value based on type T
+                T rainValue;
+                if constexpr (std::is_same<T, Rpp8u>::value)
+                {
+                    rainValue = static_cast<T>(200);
+                }
+                else if constexpr (std::is_same<T, Rpp8s>::value)
+                {
+                    rainValue = static_cast<T>(72);
+                }
+                else
+                {
+                    rainValue = static_cast<T>(200 * ONE_OVER_255);
+                }
+
+                *rainLayerTemp = rainValue;
+                
+                if (srcDescPtr->c > 1)
+                {
+                    *(rainLayerTemp + srcDescPtr->strides.cStride) = rainValue;
+                    *(rainLayerTemp + 2 * srcDescPtr->strides.cStride) = rainValue;
+                }
+            }
+        }
+    }
+
+    int globalThreads_x = (dstDescPtr->strides.hStride + 7) >> 3;
+    int globalThreads_y = dstDescPtr->h;
+    int globalThreads_z = handle.GetBatchSize();
```

Comment:
For batchsize
Please ensure to get the value from dstDescPtr->n instead of handle.GetBatchSize() function as discussed before

Reply:
Done